pkg/controller/stacks/install/installjob.go

```diff
@@ -279,13 +281,78 @@ func (jc *stackInstallJobCompleter) createJobOutputObject(ctx context.Context, o
 		"apiVersion", obj.GetAPIVersion(),
 		"kind", obj.GetKind(),
 	)
-	if err := jc.client.Create(ctx, obj); err != nil && !kerrors.IsAlreadyExists(err) {
-		return errors.Wrapf(err, "failed to create object %s from job output %s", obj.GetName(), job.Name)
+
+	if err := jc.client.Create(ctx, obj); err != nil {
+		if !kerrors.IsAlreadyExists(err) {
+			return errors.Wrapf(err, "failed to create object %s from job output %s", obj.GetName(), job.Name)
+		}
+
+		if !isCRD(obj) {
+			return nil
+		}
+
+		if err := jc.replaceCRD(ctx, obj); err != nil {
+			return errors.Wrapf(err, "can not replace existing CRD")
+		}
 	}
 
 	return nil
 }
 
+func (jc *stackInstallJobCompleter) replaceCRD(ctx context.Context, obj *unstructured.Unstructured) error {
+	existing := &apiextensions.CustomResourceDefinition{}
+	nsn := types.NamespacedName{
+		Namespace: obj.GetNamespace(),
+		Name:      obj.GetName(),
+	}
+
+	if err := jc.client.Get(ctx, nsn, existing); err != nil {
+		return errors.Wrapf(err, "failed to fetch existing crd")
+	}
+
+	if meta.WasDeleted(existing) {
+		return errors.New("existing crd has not been deleted")
+	}
+
+	crd, err := convertToCRD(obj)
+	if err != nil {
+		return errors.Wrapf(err, "failed to convert unstructured crd from job output")
+	}
+
+	if !crdIsVersionsInclusive(existing, crd) {
+		return errors.New("replacement crd removes crd versions")
+	}
+
+	// TODO(displague) REVIEWER QUESTION:
+	// capture unpacked crd labels/annotations and give them
+	// highter priority than existing labels/annotations?
```

Comment:
The unpacking process does not add any labels to CRDs, currently.  However, a CRD may have included labels within the package.

When we want to replace an existing CRD with one from a package (because it provides all of the old CRD versions and then some), we want to persist the existing labels (which denote the various stacks that share the responsibility of providing this CRD).

Annotations on the other hand, represent the UI metadata contained in the package.

The behavior of this PR is:
1. use the unpack annotations
2. merge any existing annotations from the existing CRD

It may be better (necessary?) to:
1. use the unpack annotations
2. merge any non-conflicting annotations from the existing CRD

Does it make more sense to:
1. use the unpack annotations
2. capture existing CRD annotations
3. remove from the existing CRD any annotations know to crossplane/stackmanager
2. merge the existing CRD annotations onto the new CRD annotations